### PR TITLE
Issue 345 smart imports drishya

### PR DESCRIPTION
**Pull request by drishyatm**
This PR is to change the imports /smart imports . Imports like TestRail, Tesults and Browserstack only can be called or used when the flag is set to true . Removed these imports from the beginning of the files ! 

Changed Files 
page_objects/Base_Page.py
page_objects/Mobile_Base_Page.py
conftest.py

How to Test : 

Switch to the branch 
Fill in the conf file with the configuration for TestRail, Tesults , Browserstack 
 validate still the test runs with the changes of import

### CODE REVIEW

File: page_objects/Base_Page.py [hunk 1]
```diff
@@ -6,13 +6,10 @@
 from selenium.webdriver.support.ui import WebDriverWait
 from selenium.webdriver.support import expected_conditions as EC
 from selenium.webdriver.common.action_chains import ActionChains
-import unittest,time,logging,os,inspect,pytest
+import unittest,time,logging,os,inspect
```
Comment (shivahari) [inline page_objects/Base_Page.py:9]: Can you check if we need the `unittest` module and remove it if we are not using it?

File: page_objects/Base_Page.py [hunk 2]
```diff
@@ -124,6 +122,7 @@ def get_current_driver(self):
     def register_testrail(self):
         "Register TestRail with Page"
         self.testrail_flag = True
+        from utils.Test_Rail import Test_Rail  # pylint: disable=import-error,import-outside-toplevel
         self.tr_obj = Test_Rail()
```
Comment (shivahari) [inline page_objects/Base_Page.py:126]: Can you rename this variable to `self.testrail_object`?

File: page_objects/Base_Page.py [hunk 3]
```diff
@@ -133,10 +132,13 @@ def set_test_run_id(self,test_run_id):
     def register_tesults(self):
         "Register Tesults with Page"
         self.tesults_flag = True
+        from utils import Tesults # pylint: disable=import-error,import-outside-toplevel
+        self.tesult_objects = Tesults
```
Comment (shivahari) [inline page_objects/Base_Page.py:136]: Can you rename this variable to `self.testult_object`?

**Review by shivahari — CHANGES_REQUESTED** [review]

File: page_objects/Base_Page.py (hunk 3 above)
Comment (drishyatm) [reply page_objects/Base_Page.py:136]: The name is "Tesults" not "testult". May I know the reason to change the name ?

File: page_objects/Base_Page.py (hunk 2 above)
Comment (drishyatm) [reply page_objects/Base_Page.py:126]: self.tr_obj = Test_Rail()
 
This was existing before and not a new change .  I think it follows as TestRail as tr . If we proceed with this update, we might need to rename numerous locations to preserve consistency throughout the framework Basepage, Mobile Base Page, and setup_testrail.

File: page_objects/Base_Page.py (hunk 1 above)
Comment (drishyatm) [reply page_objects/Base_Page.py:9]: class Base_Page(Borg,unittest.TestCase):
    "Page class that all page models can inherit from"

Unit test is being used inside the page .

File: page_objects/Base_Page.py (hunk 3 above)
Comment (shivahari) [reply page_objects/Base_Page.py:136]: Sorry. I meant `self.tesult_object` instead of `self.tesult_objects`

File: page_objects/Base_Page.py (hunk 1 above)
Comment (shivahari) [reply page_objects/Base_Page.py:9]: We don't need `unittest.TestCase` Drishya. We can remove it, we are using `pytest`

File: page_objects/Base_Page.py (hunk 2 above)
Comment (shivahari) [reply page_objects/Base_Page.py:126]: Yeah Drishya, we seem to be persisting with the bad variable name.  Let's change it as part of this PR.

Comment (shivahari) [reply page_objects/Base_Page.py:126]: I ran a `grep` to find out where we have used `self.tr_obj`, we seem to be using it in:
- `./page_objects/Base_Page.py:        self.tr_obj = Test_Rail()`
- `./page_objects/Base_Page.py:            self.tr_obj.update_testrail(case_id,test_run_id,result_flag,msg=msg)`
- `./page_objects/Mobile_Base_Page.py:        self.tr_obj = Test_Rail()`
- `./page_objects/Mobile_Base_Page.py:            self.tr_obj.update_testrail(case_id,test_run_id,result_flag,msg=msg)`

File: page_objects/Base_Page.py (hunk 1 above)
Comment (drishyatm) [reply page_objects/Base_Page.py:9]: pytest I removed as it was not using in this module anywhere . Do you mean I should include pytest and remove unittest?

File: page_objects/Base_Page.py (hunk 2 above)
Comment (drishyatm) [reply page_objects/Base_Page.py:126]: Done!

File: page_objects/Base_Page.py (hunk 3 above)
Comment (drishyatm) [reply page_objects/Base_Page.py:136]: done!

**Review by shivahari — APPROVED** [review]
The changes look good. Please **squash** your commits, **rebase** with `master` branch and merge your changes.

**Pull request merged**